### ISSUE TITLE
torch.nn.CTCLoss not reproducible on GPU

### ISSUE THREAD

**Issue by jamfly** [speechbrain/speechbrain#943]
A previous [repot](https://github.com/pytorch/pytorch/issues/17798) indicated that native torch CTC loss might not reproducible on GPU under certain conditions. I found that [ESPnet](https://github.com/espnet/espnet/blob/master/espnet/nets/pytorch_backend/ctc.py#L68) also encounter the same issue and they patched it.

Don't know if this will influence the performance on different devices.

Thank you

**Comment by mravanelli** [comment]
Any idea about how we can fix that? Not sure this is a problem though

**Comment by anautsch** [comment]
The PyTorch documentation confirms this for ["CUDA version is 10.2 or greater"](https://pytorch.org/docs/master/generated/torch.use_deterministic_algorithms.html?highlight=use_deterministic_algorithms#torch.use_deterministic_algorithms). The issue is linked to #751 which we will tackle over the next weeks.

**Comment by vadimkantorov** [comment]
> A previous [repot](https://github.com/pytorch/pytorch/issues/17798) indicated that native torch CTC loss might not reproducible on GPU under certain conditions. I found that [ESPnet](https://github.com/espnet/espnet/blob/master/espnet/nets/pytorch_backend/ctc.py#L68) also encounter the same issue and they patched it.
> 
> Don't know if this will influence the performance on different devices.
> 
> Thank you

EspNet solution is probably not working: https://github.com/espnet/espnet/issues/4412
